src/ucp/tag/rndv.c

```diff
@@ -637,11 +725,17 @@ static void ucp_rndv_prepare_zcopy_send_buffer(ucp_request_t *sreq, ucp_ep_h ep)
         (ucp_ep_get_am_lane(ep) != ucp_ep_get_tag_lane(ep))) {
         ucp_request_send_buffer_dereg(sreq, ucp_ep_get_tag_lane(sreq->send.ep));
         sreq->send.state.dt.contig.memh = UCT_MEM_HANDLE_NULL;
-    } else if ((ucp_ep_is_rndv_lane_present(ep)) &&
-        (ucp_ep_get_am_lane(ep) != ucp_ep_get_rndv_get_lane(ep))) {
+        ucp_request_clear_rails(&sreq->send.state);
+    } else if ((ucp_ep_is_rndv_lane_present(ep, 0)) &&
+        (ucp_ep_get_am_lane(ep) != ucp_ep_get_rndv_get_lane(ep, 0))) {
```

Comment:
indentation (thought it was not added by you:)

Reply:
will fix